src/graphcast_agent/mod.rs

```diff
@@ -47,6 +49,87 @@ pub const REGISTRY_SUBGRAPH: &str =
 /// A constant defining the goerli network subgraph endpoint.
 pub const NETWORK_SUBGRAPH: &str = "https://gateway.testnet.thegraph.com/network";
 
+#[derive(Debug, thiserror::Error)]
+pub enum ConfigError {
+    #[error("Validate the input: {0}")]
+    ValidateInput(String),
+    #[error("Unknown error: {0}")]
+    Other(anyhow::Error),
+}
+
+pub async fn validate_set_up(
+    wallet_key: &str,
+    registry_subgraph: &str,
+    network_subgraph: &str,
+    graph_node_endpoint: &str,
+) -> Result<(), ConfigError> {
+    let wallet = build_wallet(wallet_key).map_err(|e| {
+        ConfigError::ValidateInput(format!(
+            "Invalid key to wallet, use private key or mnemonic: {e}"
+        ))
+    })?;
+    let graphcast_id = graphcast_id_address(&wallet);
+    // TODO: Implies invalidity for both graphcast id and registry, maybe map_err more specifically
+    let indexer = query_registry_indexer(registry_subgraph.to_string(), graphcast_id)
+        .await
+        .map_err(|e| ConfigError::ValidateInput(format!("The registry subgraph did not contain an entry for the Graphcast ID to Indexer: {e}")))?;
+    debug!("Resolved indexer identity: {}", indexer);
+
+    let stake = query_network_subgraph(network_subgraph.to_string(), indexer)
+        .await
+        .map_err(|e| {
+            ConfigError::ValidateInput(format!(
+                "The network subgraph must contain an entry for the Indexer stake: {e}"
+            ))
+        })?
+        .indexer_stake();
+    debug!("Resolved indexer stake: {}", stake);
+
+    let _ = get_indexing_statuses(graph_node_endpoint.to_string())
+        .await
+        .map_err(|e| {
+            ConfigError::ValidateInput(format!(
+                "Graph node endpoint must be able to serve indexing statuses query: {e}"
+            ))
+        })?;
+    Ok(())
+}
+
+pub struct GraphcastAgentConfig {
+    pub wallet_key: String,
+    pub radio_name: &'static str,
+    pub registry_subgraph: String,
+    pub network_subgraph: String,
+    pub graph_node_endpoint: String,
+    pub boot_node_addresses: Vec<String>,
```

Comment:
I think `boot_node_addresses` should already be converted to multiaddr at this point

Reply:
I don't think we should leave that to Radios, since they don't even need the `waku` dependency. It seems more logical to me that the Radios only deal with strings and we construct a Waku topic on the SDK side

Reply:
That makes sense. I think in this case, we should still restrict the type for boot_node_addresses to `Vec<Multiaddr>`, but allow the input for `GraphcastAgentConfig::new` to be `Vec<String>` and convert the input automatically before creating `GraphcastAgentConfig`